libsql-server/src/namespace/meta_store.rs

```diff
@@ -322,18 +374,40 @@ impl MetaStore {
         tracing::debug!("removing namespace `{}` from meta store", namespace);
 
         let mut guard = self.inner.lock();
-        guard.conn.execute(
-            "DELETE FROM namespace_configs WHERE namespace = ?",
-            [namespace.as_str()],
-        )?;
-        if let Some(sender) = guard.configs.remove(&namespace) {
+        let r = if let Some(sender) = guard.configs.get(&namespace) {
             tracing::debug!("removed namespace `{}` from meta store", namespace);
             let config = sender.borrow().clone();
+            let tx = guard.conn.transaction()?;
+            if let Some(shared_schema) = config.config.shared_schema_name.as_deref() {
+                tx.execute(
+                    "DELETE FROM shared_schema_links WHERE shared_schema_name = ? AND db_name = ?",
+                    [shared_schema, namespace.as_str()],
+                )?;
+            }
+            if config.config.is_shared_schema {
+                let res: rusqlite::Result<u32> = tx.query_row(
+                    "SELECT COUNT(*) FROM shared_schema_links WHERE shared_schema_name = ?",
```

Comment:
this should not be necessary if:
- we enable foreign keys on the connection (`PRAGMA foreign_keys=ON`)
- add a deletion restriction on the table
    - `FOREIGN KEY (shared_schema_name) REFERENCES namespace_configs (namespace) ON DELETE RESTRICT`
    - `FOREIGN KEY (db_name) REFERENCES namespace_configs (namespace) ON DELETE RESTRICT`

Reply:
This is true. I thought about removing the check but ended up leaving it just to be able to give user a better error message with info how many existing namespaces still link to the schema. Will remove.